source/SpinalHDL/Simulation/bootstraps.rst

```diff
@@ -79,18 +79,18 @@ This function will return a ``SimCompiled`` instance.
 
 On this ``SimCompiled`` instance you can run your simulation with the following functions:
 
-.. list-table::
-   :header-rows: 1
-
-   * - Syntax
-     - Description
-   * - ``doSim[(simName[, seed])]{dut => ...}``
-     - Run the simulation until the main thread is done (doesn't wait on forked threads) or until all threads are stuck
-   * - ``doSimUntilVoid[(simName[, seed])]{dut => ...}``
-     - Run the simulation until all threads are done or stuck
+``doSim[(simName[, seed])]{dut => /* main stimulus code */}``
+  Run the simulation until the main thread runs to completion and exits/returns.
+  It will detect and report an error if the simulation gets fully stuck. As long as
+  a e.g. a clock is running the simulation can continue forever, it is therefore recommended
+  to use ``simTimeout(cycles)`` to limit the possible runtime.
 
+``doSimUntilVoid[(simName[, seed])]{dut => ...}``
+  Run the simulation until it is ended by calling either ``simSuccess()`` or ``simFailure()``.
+  The main stimulus thread can continue or exit early, as long as there are events to process,
```

Comment:
I am not sure I understand: does `simSuccess` stop simulation or not?

Reply:
`simSuccess` will stop the simulation

Reply:
So I find this sentence being just after the end functions confusing "The main stimulus thread can continue or exit early".

What about telling that another way?

* the end of the block does not stop simulation
* simulation stops only when no more events to process, or called one of the two end functions